lib/folio_record.rb

```diff
@@ -45,6 +41,32 @@ def hrid
     instance['hrid']
   end
 
+  def marc_record
+    @marc_record ||= begin
+      record ||= MARC::Record.new_from_hash(stripped_marc_json || instance_derived_marc_record)
+
+      # if 590 with Bound-with related subfields are present, return the record as is
+      field590 = record.find { |f| f.tag == '590' }
+      subfield_a = field590&.find { |sf| sf.code == 'a' }
+      subfield_c = field590&.find { |sf| sf.code == 'c' }
+      unless field590 && subfield_a && subfield_c
+        # if 590 or one of its Bound-with related subfields is missing, and FOLIO says this record is Bound-with, append the relevant data from FOLIO
+        parents ||= bound_with_parents
+        # if Bound-with parents are found, edit the marc record
+        if parents && !parents.empty?
```

Comment:
Is something like this equivalent (and easier to read?)?

```suggestion
        if parents&.any?
```